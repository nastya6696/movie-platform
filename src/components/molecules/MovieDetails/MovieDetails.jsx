import {Header} from "../../atoms";

import PulpFiction from '../../../assets/images/pulpFiction.png';
import BohemianRhapsody from '../../../assets/images/bohemianRhapsody.png';
import Avengers from '../../../assets/images/avengers.png';
import KillBill from '../../../assets/images/killBill.png';
import Inception from '../../../assets/images/inception.png';
import DogsReservoir from '../../../assets/images/dogsReservoir.png';

import styles from './styles.module.scss';

const ImgSrc = {
  PulpFiction,
  BohemianRhapsody,
  Avengers,
  KillBill,
  Inception,
  DogsReservoir
}

export const MovieDetails = ({details, handleSearchBtnClick}) => {
  const {title, year, type, imgName, overview, duration, rating} = details;

  return (
    <div className={styles.movieDetails}>
      <Header btnClickHandler={handleSearchBtnClick} btnInfo={{ type: 'search' }} />
      <div className={styles.container}>
        <img alt={title} src={ImgSrc[imgName]} />
        <div className={styles.details}>
          <h1 className={styles.title}>{title.toUpperCase()}</h1>
          <span className={styles.rating}>{rating}</span>
          <p className={styles.genres}>{type}</p>
          <div className={styles.releaseInfo}>
            <span>{year}</span>
            <span>{duration}</span>
          </div>
          <p className={styles.overview}>{overview}</p>
        </div>
      </div>
    </div>
  )
}
