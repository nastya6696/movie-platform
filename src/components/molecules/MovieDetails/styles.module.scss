@import 'src/assets/scss/common';

.movieDetails {
  height: 606px;
  background-color: $black;
  padding: 35px 60px 29px;
}

.container {
  display: flex;
}

.details {
  margin-left: 60px;
  color: $white;
  position: relative;
}

.title {
  @include fontSizes(40px, 48px);
  font-weight: 300;
  display: inline-block;
  margin-right: 25px;
}

.rating {
  display: inline-block;
  width: 60px;
  height: 60px;
  border: 1px solid $white;
  border-radius: 50%;
  padding-top: 18px;
  text-align: center;
  @include fontSizes(20px, 24px);
  font-weight: 100;
  position: absolute;
  top: -8px;
}

.genres {
  margin-top: 9px;
  @include fontSizes(14px, 17px);
  color: $white-50-transparent;
}

.releaseInfo {
  margin-top: 30px;
  color: $red;
  @include fontSizes(24px, 29px);

  span:first-of-type {
    margin-right: 50px;
  }
}

.overview {
  margin-top: 30px;
  color: $white-50-transparent;
  @include fontSizes(20px, 24px);
}
