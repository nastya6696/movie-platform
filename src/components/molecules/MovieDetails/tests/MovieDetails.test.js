import {create} from 'react-test-renderer';
import {MovieDetails} from "../MovieDetails";

describe('MovieDetails', () => {
  it('should create simple MovieDetails component', () => {
    const handleSearchBtnClick = jest.fn();
    const details = {
      title: 'tests title',
      release_date: '2021-01-01',
      genres: ['Drama'],
      poster_path: 'https://posterpath.com',
      overview: 'overview',
      runtime: 108,
      vote_average: 9
    };

    const component = create(<MovieDetails handleSearchBtnClick={handleSearchBtnClick} details={details}/>);

    expect(component).toMatchSnapshot();
  })
})
