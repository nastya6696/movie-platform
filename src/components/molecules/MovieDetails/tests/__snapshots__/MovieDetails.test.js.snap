// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should create simple MovieDetails component 1`] = `
<div
  className="movieDetails"
>
  <div
    className="header"
  >
    <div
      className="logoText"
    >
      <strong
        className="boldText"
      >
        netflix
      </strong>
      roulette
    </div>
    <button
      className="search"
      onClick={[MockFunction]}
    >
      
    </button>
  </div>
  <div
    className="container"
  >
    <img
      alt="test title"
      className="img"
      src="https://posterpath.com"
    />
    <div
      className="details"
    >
      <h1
        className="title"
      >
        TEST TITLE
      </h1>
      <span
        className="rating"
      >
        9
      </span>
      <p
        className="genres"
      >
        Drama
      </p>
      <div
        className="releaseInfo"
      >
        <span>
          2021
        </span>
        <span>
          108
        </span>
      </div>
      <p
        className="overview"
      >
        overview
      </p>
    </div>
  </div>
</div>
`;
